full/src/test/java/apoc/systemdb/SystemDbTest.java

```diff
@@ -74,5 +101,74 @@ public void testWriteStatements() {
+
+        final String constraintForUuid = "CREATE CONSTRAINT IF NOT EXISTS ON (n:Person) ASSERT n.alpha IS UNIQUE;";
+        db.executeTransactionally(constraintForUuid);
+        final String uuidStatement = "CALL apoc.uuid.install('Person', {uuidProperty: \"alpha\",addToSetLabels: true});";
+        db.executeTransactionally(uuidStatement);
+
+        final String dvStatement = "CALL apoc.dv.catalog.add('dvName',{name:\"dvName\",url:\"file://myUrl\",desc:\"person's details\",labels:[\"Person\"],query:\"map.name = $name and map.age = $age\",params:[\"$name\",\"$age\"],type:\"CSV\"})";
+        db.executeTransactionally(dvStatement);
+
+        TestUtil.testCallEmpty(db, "CALL apoc.systemdb.export.metadata", Collections.emptyMap());
+        
+        assertEquals(cleanStatement(constraintForUuid), readLines("metadata.Uuid.schema.neo4j.cypher"));
+        assertEquals(cleanStatement(uuidStatement), readLines("metadata.Uuid.neo4j.cypher"));
+        assertEquals(cleanStatement(triggerOne, triggerTwo, pauseTrigger), readLines("metadata.Trigger.neo4j.cypher"));        
+        assertEquals(cleanStatement(declareProcedure, declareStatementFromProcedure), readLines("metadata.CypherProcedure.neo4j.cypher"));
+        assertEquals(cleanStatement(declareFunction, declareStatementFromFunction), readLines("metadata.CypherFunction.neo4j.cypher"));
+        assertEquals(cleanStatement(dvStatement), readLines("metadata.DataVirtualizationCatalog.neo4j.cypher"));
+        
+        // -- with config and uuid constrain dropped
+        db.executeTransactionally("DROP CONSTRAINT ON (p:Person) ASSERT p.alpha IS UNIQUE");
+        TestUtil.testCallEmpty(db, "CALL apoc.systemdb.export.metadata($config)", 
+                Map.of("config", Map.of(FILENAME_KEY, "custom", FEATURES_KEY, Set.of(ExportMetadata.Type.Uuid.name()))));
+
+        db.executeTransactionally("CALL apoc.uuid.removeAll");
+        db.executeTransactionally("CALL apoc.trigger.removeAll");
+
+        assertEquals(cleanStatement(constraintForUuid), readLines("custom.Uuid.schema.neo4j.cypher"));
+        assertEquals(cleanStatement(uuidStatement), readLines("custom.Uuid.neo4j.cypher"));
+    }
 
+    private Set<String> cleanStatement(String ...statements) {
+        return Set.of(statements).stream()
+                .map(this::statementCleaning)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<String> readLines(String fileName) {
+        try {
+            return FileUtils.readLines(new File(directory, fileName), StandardCharsets.UTF_8).stream()
+                    .map(this::statementCleaning)
+                    .collect(Collectors.toSet());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private String statementCleaning(String string) {
+        return string.replaceAll("\\s*:\\s*", ":").replaceAll("\\s*,\\s*", ",");
```

Comment:
non ho capito a cosa serve questo???

Reply:
Per togliere gli spazi ad esempio da questo `CALL apoc.trigger.add('beta','RETURN 1', null, {params:{}});` a questo `CALL apoc.trigger.add('beta','RETURN 1',null,{params:{}});`, così se metto nella dichiarazione della procedura degli spazi in più dopo la virgola, l'assertion non spacca.
Volendo potrei anche fare tutti gli statement con tutto attaccato?